### PR TITLE
improve test 1864 by using latin1 encoded strings

### PR DESCRIPTION
**Pull request by shrektan**
So it can be tested on all the platforms. It improves the test for PR #2566.

### CODE REVIEW

File: inst/tests/tests.Rraw [hunk 1]
```diff
@@ -11146,16 +11146,20 @@ DT = data.table(col1 = c(1,1,1, 2,2,2), col2 = c(2,2,2,1,1,1), ID = c(rep(1,3),
...
+utf8_strings <- c("\u00e7ile", "fa\u00e7ile", "El. pa\u00c5\u00a1tas", "\u00a1tas", "\u00de")
+latin1_strings <- iconv(utf8_strings, from = "UTF-8", to = "latin1")
+mixed_strings <- c(utf8_strings, latin1_strings)
+DT1 <- data.table(x = mixed_strings, y = c(latin1_strings, utf8_strings), z = 1:10)
+DT2 <- copy(DT1)
+setkey(DT1, x)
+setkey(DT2, y)
+test(1864.1, DT1$x, sort(c(utf8_strings, utf8_strings), method = "radix"))
+test(1864.2, DT2$y, sort(c(utf8_strings, utf8_strings), method = "radix"))
+test(1864.3, DT1[J(utf8_strings)], DT1[J(latin1_strings)])
```
Comment (mattdowle) [inline inst/tests/tests.Rraw]: Imagine that, for reason of a strange bug, both `DT1[J(utf8_strings)]` and `DT1[J(latin1_strings)]` returned NULL.  The test would still pass. I like this pattern for that reason : 
test(1864.3,  DT1[J(utf8_strings)],  ans<- data.table( .. known answer .. ) )
test(1864.4,  DT1[J(latin1_strings)], ans)

Comment (mattdowle) [reply inst/tests/tests.Rraw]: Perhaps use the integers in `z` to construct `ans`.

Comment (mattdowle) [reply inst/tests/tests.Rraw]: Otherwise, great!

Comment (shrektan) [reply inst/tests/tests.Rraw]: You're right. Using an explicit answer is indeed better.

**Pull request merged**